client/src/components/AdminContainer/index.js

```diff
@@ -48,15 +49,19 @@ const AdminContainer = ({ children, buttonContent, buttonRoute, logout }) => {
                 </li>
               </ul>
               <div className="admin-side-btn">
-                <Button
-                  className="logout-btn"
-                  type="primary"
-                  onClick={() => {
-                    logout();
-                  }}
-                >
-                  Logout
-                </Button>
+                <LogoutContext.Consumer>
+                  {({ logout }) => (
+                    <Button
+                      className="logout-btn"
+                      type="primary"
+                      onClick={() => {
```

Comment:
@Mu7ammadAbed I think you can remove the arrow function and use logout directly, like this: `onClick={logout}`.  That's slightly better because then this class component won't create a new function in memory every time it re-renders.